File: servers/Services/AcasService/Models/Subject.cs
using System.ComponentModel.DataAnnotations;

namespace AcasService.Models;

public class Subject
{
    [Key]
    public string Id { get; set; } = string.Empty;
    
    [Required]
    public string SubjectCode { get; set; } = string.Empty;
    
    [Required]
    public string SubjectName { get; set; } = string.Empty;
    
    public string Description { get; set; } = string.Empty;
    
    [Required]
    public string CreatedBy { get; set; } = string.Empty;
    
    public bool IsDeleted { get; set; }
    
    public DateTime CreatedDate { get; set; }
    
    public DateTime? UpdatedDate { get; set; }
}

File: mobile_app/lib/features/models/subject.dart
class Subject {
  final String id;
  final String subjectCode;
  final String subjectName;
  final bool isDeleted;

  Subject({
    required this.id,
    required this.subjectCode,
    required this.subjectName,
    required this.isDeleted,
  });

  factory Subject.fromJson(Map<String, dynamic> json) {
    return Subject(
      id: json['id'] ?? '',
      subjectCode: json['subjectCode'] ?? '',
      subjectName: json['subjectName'] ?? '',
      isDeleted: json['isDeleted'] == true,
    );
  }
}
